UnitWorld_shared/shared/communication/game/CommunicatedSinguity.cpp

```diff
@@ -0,0 +1,69 @@
+#include "CommunicatedSinguity.h"
+
+#include <nlohmann/json.hpp>
+
+using namespace uw;
+
+CommunicatedSinguity::CommunicatedSinguity(const xg::Guid& singuityId, const xg::Guid& playerId, const CommunicatedVector2D& position, const CommunicatedVector2D& speed, const Option<CommunicatedVector2D>& destination) :
+    _singuityId(singuityId),
+    _playerId(playerId),
+    _position(position),
+    _speed(speed),
+    _destination(destination)
+{}
+
+std::string CommunicatedSinguity::toJson() const
+{
+    const auto singuityId = _singuityId.str();
+    const auto playerId = _playerId.str();
+    const auto position = nlohmann::json::parse(_position.toJson());
+    const auto speed = nlohmann::json::parse(_speed.toJson());
+    const auto destination = _destination.map<nlohmann::json>([](const auto& destination) { return nlohmann::json::parse(destination.toJson()); }).getOrElse(nlohmann::json("none"));
+
+    nlohmann::json jsonData = {
+        {"id", singuityId},
+        {"playerId", playerId},
+        {"position", position},
+        {"speed", speed},
+        {"destination", destination}
```

Comment:
All "keys" (`"id"`, `"playerId"`, `"position"`, `"speed"`, `"destination"`) are used at many places. Maybe use static constants instead? If you mind to change the name, only a single line of code would need to change.

Reply:
will do in a later PR, much of these changed later in the development